api/alembic/versions/c2cd7a585bbd_recompute_combustible_area.py

```diff
@@ -0,0 +1,67 @@
+"""recompute combustible area
+
+Revision ID: c2cd7a585bbd
+Revises: 8635552697ad
+Create Date: 2024-05-15 08:32:04.942965
+
+"""
+from alembic import op
+import sqlalchemy as sa
+import geoalchemy2
+from sqlalchemy.orm.session import Session
+from app.auto_spatial_advisory.calculate_combustible_land_area import calculate_combustible_area_by_fire_zone, get_fuel_types_from_db
+
+# revision identifiers, used by Alembic.
+revision = 'c2cd7a585bbd'
+down_revision = '8635552697ad'
+branch_labels = None
+depends_on = None
+
+
+shape_type_table = sa.Table('advisory_shape_types', sa.MetaData(),
+                            sa.Column('id', sa.Integer),
+                            sa.Column('name', sa.String))
+
+shape_table = sa.Table('advisory_shapes', sa.MetaData(),
+                       sa.Column('id', sa.Integer),
+                       sa.Column('source_identifier', sa.String),
+                       sa.Column('shape_type', sa.Integer),
+                       sa.Column('geom', geoalchemy2.Geometry))
+
+
+def get_fire_zone_unit_shape_type_id(session: Session):
+    statement = shape_type_table.select().where(shape_type_table.c.name == 'fire_zone_unit')
+    result = session.execute(statement).fetchone()
+    return result.id
+
+
+def get_fire_zone_units(session: Session, fire_zone_type_id: int):
+    statement = shape_table.select().where(shape_table.c.shape_type == fire_zone_type_id)
+    result = session.execute(statement).fetchall()
+    return result
+
+
+def upgrade():
+    session = Session(bind=op.get_bind())
+
+    with get_fuel_types_from_db() as fuel_types:
```

Comment:
@conbrad @dgboss Any idea if I can use data that was inserted into the database in an earlier migration? This migration works fine locally, with an existing db, but not in dev. Might be a gdal issue too

Reply:
You can try to pull a prod backup and test locally, matching gdal 3.4 might be tricky though.

Reply:
You can't import a function/module that accesses the db. You can access an existing table in a migration, but you have to 'redefine' it as you have done for the shape_table and shape_types_table. You'll have to duplicate the code of the `get_fuel_types_from_db` function in this migration.

Reply:
Oh, I see you're using gdal in `get_fuel_types_from_db`. You'll have to use a different approach for pulling our the fuel types from the db.

Reply:
Thanks guys. Not being able to import anything that accesses the db makes sense, any idea why this works fine locally?